src/components/DataProvider/index.js

```diff
@@ -592,4 +596,6 @@ DataProvider.defaultProps = {
   yAxisWidth: 50,
   yDomain: null,
   ySubDomain: null,
+  isXSubDomainSticky: false,
+  limitXSubDomain: null,
```

Comment:
What if we made the default `xSubDomain => xSubDomain`? Then we could skip the conditionals, and it's self-documenting.